app/javascript/src/javascripts/post_flags.js

```diff
@@ -1,17 +1,176 @@
+  // Use a live collection so newly-checked inputs are reflected
+  const radioButtons = document.getElementsByName("post_flag[reason_name]");
+  // Find the form by traversing up from the note field, or fallback to the first form in the flag-dialog-body
+  let form = null;
+  if (noteField) {
+    form = noteField.closest("form");
+  } else {
+    const dialogBody = document.querySelector(".flag-dialog-body");
+    if (dialogBody) {
+      form = dialogBody.querySelector("form");
+    }
+  }
+  let errorMessage = null;
+
+  // Parse dataset boolean values robustly. Treat presence or values like "", "true", "1", "yes" as true.
+  function parseDatasetBoolean (val) {
+    if (val === undefined || val === null) return false;
+    const v = String(val).trim().toLowerCase();
+    if (v === "" || v === "true" || v === "1" || v === "yes") return true;
+    return false;
+  }
+
+  function updateNoteRequired () {
+    try {
+      if (!noteField) return;
+      const selected = document.querySelector("input[name=\"post_flag[reason_name]\"]:checked");
+      const label = noteField.closest(".flag-notes")?.querySelector("label");
+      if (selected && parseDatasetBoolean(selected.dataset.requireExplanation)) {
+        noteField.required = true;
+        if (label) {
+          // Add a small indicator span instead of replacing the label content
+          let indicator = label.querySelector(".required-indicator");
+          if (!indicator) {
+            indicator = document.createElement("span");
+            indicator.className = "required-indicator";
+            indicator.textContent = " (required)";
+            label.appendChild(indicator);
+          }
+        }
+      } else {
+        noteField.required = false;
+        if (label) {
+          const indicator = label.querySelector(".required-indicator");
+          if (indicator) indicator.remove();
+        }
+      }
+    } catch (err) {
+      console.error("PostFlags.updateNoteRequired error", err);
+    }
+  }
+
+  function showError (msg) {
+    try {
+      if (!noteField) return;
+      if (!errorMessage) {
+        errorMessage = document.createElement("div");
+        errorMessage.className = "flag-note-error error-text";
+        noteField.closest(".flag-notes")?.appendChild(errorMessage);
+      }
+      errorMessage.textContent = msg;
+    } catch (err) {
+      console.error("PostFlags.showError error", err);
+    }
+  }
+
+  function clearError () {
+    if (errorMessage) errorMessage.textContent = "";
+  }
+
+  function isNoteRequiredAndEmpty () {
+    try {
+      if (!noteField) return false;
+      const selected = document.querySelector("input[name=\"post_flag[reason_name]\"]:checked");
+      return selected && parseDatasetBoolean(selected.dataset.requireExplanation) && noteField.value.trim() === "";
+    } catch (err) {
+      console.error("PostFlags.isNoteRequiredAndEmpty error", err);
+      return false;
+    }
```

Comment:
The code contains a lot of try-catch statements like this one that catch nothing. None of the functions inside the try body can actually throw, so this just bloats the code.